src/validate.js

```diff
@@ -0,0 +1,266 @@
+const catalogInfoSchema = require('../catalogInfoSchema/CatalogInfo.schema.json');
+const apiSchema = require('../catalogInfoSchema/API.v1alpha1.schema.json');
+const componentSchema = require('../catalogInfoSchema/Component.v1alpha1.schema.json');
+const domainSchema = require('../catalogInfoSchema/Domain.v1alpha1.schema.json');
+const resourceSchema = require('../catalogInfoSchema/Resource.v1alpha1.schema.json');
+const systemSchema = require('../catalogInfoSchema/System.v1alpha1.schema.json');
+const entitySchema = require('../catalogInfoSchema/Entity.schema.json');
+
+const SCHEMA_URL = 'https://github.com/im-open/validate-catalog-info-file/blob/main/catalogInfoSchema/CatalogInfo.schema.json';
+const API_SCHEMA_URL = 'https://github.com/im-open/validate-catalog-info-file/blob/main/catalogInfoSchema/API.v1alpha1.schema.json';
+
+function setupAjvSchemaValidation(Ajv) {
+  // If allErrors is not set, the validator will stop once the first validation error occurrs
+  const ajv = new Ajv({
+    allErrors: true,
+    async: true,
+    verbose: true,
+    unicodeRegExp: false
+  });
+
+  ajv.addSchema(catalogInfoSchema, 'root');
+  ajv.addSchema(entitySchema, 'entity');
+  ajv.addSchema(apiSchema, 'api');
+  ajv.addSchema(componentSchema, 'component');
+  ajv.addSchema(domainSchema, 'domain');
+  ajv.addSchema(resourceSchema, 'resource');
+  ajv.addSchema(systemSchema, 'system');
+
+  // Schema compilation happens on the first API call, not when they are added.  So make a
+  // call here to force it to compile, so when we go to validate it doesn't need to do it.
+  ajv.getSchema('root');
```

Comment:
Why does it matter when compilation happens?

Reply:
From ajv:
>Ajv validation model is optimized for server side execution, when schema compilation happens only once and validation happens multiple times - this has a substantial performance benefit comparing with validators that interpret the schema in the process of validation.

Reply:
It can be "slow" so you want the compilation to happen once and happen before you need it so it's ready to go once you call validate.  In the grand scheme of things it probably isn't that slow and the first call might not be necessary.

Reply:
Oh does the compilation happen asynchronously? Is that what you meant by "happen before you need it"? I didn't see that in the ajv docs, but I only skimmed them.  If it were a synchronous process then I figured the compilation could happen at any time and we wouldn't lose any speed, hence my question.